src/fparser/two/Fortran2003.py

```diff
@@ -7203,102 +7208,456 @@ def match(string):
                                  string, require_cls=True)
 
 
-class Format_Specification(BracketBase):  # R1002
-    """
-    <format-specification> = ( [ <format-item-list> ] )
-    """
+class Format_Item_List(SequenceBase):  # pylint: disable=invalid-name
+    '''This class replaces the one generated by fparser. This class is
+    required as support for hollerith strings makes matching more
+    complicated.
+
+    '''
+    use_names = []
+    subclass_names = ['Format_Item']
+
+    @staticmethod
+    def match(string):
+        '''Implements the matching for a list of format items.
+
+        Supporting Hollerith strings makes it very difficult to
+        correctly split the input string into items a-priori. The
+        reason for this can be seen in the following example:
+
+        `2H,x,e2.2` is `2H,x` and `e2.2` but when split with commas
+        incorrectly gives `2H`, `x` and `e2.2`.
+
+        Further, hollerith strings could also confuse any code that
+        tried to determine whether code was inside quotes or not. For
+        example:
+
+        `2H"x,2H"x` does not mean that `x,2H` is part of a string.
+
+        The solution chosen is to match one item at a time, first
+        checking for a valid Hollerith string and then checking for
+        any other valid input.
+
+        :param str string: the string to match as a Format List.
+        :return: None if there is no match or a tuple of size 2 \
+        containing a string with a comma followed by a tuple \
+        containing a list which itself contains the matched \
+        format items.
+        :rtype: (`str`, \
+        ([:py:class:`fparser.two.Fortran2003.Format_Item`s])) or `NoneType`
+
+        '''
+        if not string:
+            return None
+        current_string = string.strip()
+        if not current_string:
+            return None
+        item_list = []
+        while current_string:
+            # Does the current item match the start of a
+            # hollerith string?
+            pattern = Hollerith_Item.match_pattern
```

Comment:
pylint complains that `pattern` is being re-defined from outer scope.

Reply:
So it is. Sorry, I missed that. Changed the name.